components/admin_console/billing/billing_subscriptions/limits.tsx

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useEffect} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {cloudFreeEnabled} from 'mattermost-redux/selectors/entities/preferences';
+import {getCloudLimits} from 'mattermost-redux/selectors/entities/cloud';
+import {getCloudLimits as getCloudLimitsAction} from 'actions/cloud';
+
+import LimitCard from './limit_card';
+
+export const GB = 1 * 8 * 1024 * 1024 * 1024;
+
+// TODO: Replace this with actual usages stored in redux,
+// that ideally are updated with a websocket event in near real time.
+const fakeUsage = {
+    files: {
+        totalStorage: Number(GB),
+    },
+    messages: {
+        history: 11000,
+    },
+    boards: {
+        cards: 300,
+    },
+    integrations: {
+        enabled: 9,
+    },
+};
+
+const Limits = (): JSX.Element | null => {
+    const isCloudFreeEnabled = useSelector(cloudFreeEnabled);
+    const cloudLimits = useSelector(getCloudLimits);
+    const dispatch = useDispatch();
+    const [requestedLimits, setRequestedLimits] = useState(false);
+
+    useEffect(() => {
+        if (isCloudFreeEnabled && !requestedLimits) {
+            dispatch(getCloudLimitsAction());
+            setRequestedLimits(true);
+        }
+    }, [isCloudFreeEnabled, requestedLimits]);
+
+    if (!isCloudFreeEnabled || !(cloudLimits?.messages?.history)) {
+        return null;
+    }
+
+    return (
+        <div className='ProductLimitsPanel'>
+            <div className='ProductLimitsPanel__title'>
+                <FormattedMessage
+                    id='workspace_limits.upgrade'
+                    defaultMessage='Upgrade to avoid {planName} data limits'
+                    values={{
+                        planName: 'Cloud Starter',
+                    }}
+                />
+            </div>
+            <div className='ProductLimitsPanel__limits'>
+                {cloudLimits?.files?.total_storage && (
+                    <LimitCard
+                        name={(
+                            <FormattedMessage
+                                id='workspace_limits.file_storage'
+                                defaultMessage='File Storage'
+                            />
+                        )}
+                        status={(
+                            <FormattedMessage
+                                id='workspace_limits.file_storage.usage'
+                                defaultMessage='{actual} of {limit} ({percent}%)'
+                                values={{
+                                    actual: `${Math.floor(fakeUsage.files.totalStorage / GB)}GB`,
+                                    limit: `${Math.floor(cloudLimits.files.total_storage / GB)}GB`,
+                                    percent: Math.floor((fakeUsage.files.totalStorage / cloudLimits.files.total_storage) * 100),
```

Comment:
Maybe some utils for this kind of calculations/formatting could help with upcoming reuse of limits.